vcd/resource_vcd_nsxt_ipsec_vpn_tunnel.go

```diff
@@ -334,39 +328,29 @@ func resourceVcdNsxtIpSecVpnTunnelUpdate(ctx context.Context, d *schema.Resource
 	// Inject ID for update
 	ipSecVpnConfig.ID = d.Id()
 
-	securityProfileHasChange := d.HasChange("security_profile_customization")
-	_, newSecurityProfile := d.GetChange("security_profile_customization")
+	_, isSet := d.GetOk("security_profile_customization")
 
-	// Security Profile Customization settings work on two different endpoints:
-	// * To set a custom security profile - there is a separate endpoint where all security profile settings can be
-	// set. After setting them, parent IPsec VPN Tunnel `SecurityType` becomes "CUSTOM".
-	// * To remove customization and switch back to NSX-T Default parameters the parent IPsec VPN Tunnel must be updated
-	// and its field 'SecurityType' must be set to 'DEFAULT'
-	if securityProfileHasChange && len(newSecurityProfile.([]interface{})) == 0 {
+	// If unset, it will set it back to default, so before updating the VPN config, we need to check if the `security_profile_customization`
+	// field is defined in the configuration
+	if isSet {
+		ipSecVpnConfig.SecurityType = "CUSTOM"
+	} else {
 		ipSecVpnConfig.SecurityType = "DEFAULT"
 	}
 
-	// At first update IPsec VPN tunnel configuration
-	// It will reset Security Profile to DEFAULT at the same shot if no customization exists in 'security_profile_customization'
 	updatedIpSecVpnConfiguration, err := existingIpSecVpnConfiguration.Update(ipSecVpnConfig)
 	if err != nil {
 		return diag.Errorf("[nsx-t ipsec vpn tunnel update] error updating NSX-T IPsec VPN Tunnel configuration '%s': %s", ipSecVpnConfig.Name, err)
 	}
 
-	// If Security Profile has change and it is being customized
-	if securityProfileHasChange && newSecurityProfile != nil {
-		ipSecTunnelProfileConfig, err := getNsxtIpSecVpnProfileTunnelConfigurationType(d)
-		if err != nil {
-			return diag.Errorf("[nsx-t ipsec vpn tunnel update] error getting NSX-T IPsec VPN Tunnel Profile: %s", err)
-		}
-
-		// To set IPsec VPN Tunnel Connection Profile - it must be updated (HTTP PUT) with all the options configured
-		if ipSecTunnelProfileConfig != nil {
-			_, err = updatedIpSecVpnConfiguration.UpdateTunnelConnectionProperties(ipSecTunnelProfileConfig)
-			if err != nil {
-				return diag.Errorf("[nsx-t ipsec vpn tunnel update] error updating NSX-T IPsec VPN Tunnel Security Profile: %s", err)
-			}
+	// Get current Security Tunnel profile, if it's "DEFAULT" in VCD, returns nil
+	ipSecTunnelProfileConfig := getNsxtIpSecVpnProfileTunnelConfigurationType(d)
 
+	// To set IPsec VPN Tunnel Connection Profile - it must be updated (HTTP PUT) with all the options configured
+	if ipSecTunnelProfileConfig != nil {
```

Comment:
Wouldn't the same check `_, isSet := d.GetOk("security_profile_customization")` as in `https://github.com/vmware/terraform-provider-vcd/pull/1072/files#diff-aadc80442709e37c8a86461fb33847073473b847e213ecdfb209f1c0a036eb06R331 be simpler to read?

Reply:
Updated here aswell. Changed the variables so they're more clear aswell.